apis/oauth/src/main/java/org/jclouds/oauth/v2/config/OAuthProperties.java

```diff
@@ -36,6 +36,35 @@
     */
    public static final String CREDENTIAL_TYPE = "jclouds.oauth.credential-type";
 
+   /**
+    * When using oauth with Azure Active Direction and Client Credentials, a "resource" must
+    * be specified as part of the request.
+    *
+    * @see <a href="https://msdn.microsoft.com/en-us/library/azure/dn645543.aspx">doc</a>
+    */
+   public static final String RESOURCE = "jclouds.oauth.resource";
+
+   /**
+    * When using oauth with Azure Active Direction and Client Credentials, an "azure-credential"
+    * is a password.  Only used for OAuth Live Tests.
+    */
+   public static final String AZURE_CREDENTIAL = "oauth.azure-credential";
+
+   /**
+    * When using oauth with Azure Active Direction and Client Credentials, an "azure-identity"
+    * is the GUID associated with an application given permssion to perform operations in
+    * Azure Resource Manager.  Only used for OAuth Live Tests.
+    */
+   public static final String AZURE_IDENTITY = "oauth.azure-identity";
+
+   /**
+    * When using oauth with Azure Active Direction and Client Credentials, the azure-endpoint
+    * is the URL https://login.microsoftonline.com/<Tenant ID>/oauth2/token, where <Tenant ID>
+    * corresponds to your Azure AD Tenant.  Only used for OAuth Live Tests.
+    */
+   public static final String AZURE_ENDPOINT = "oauth.azure-endpoint";
```

Comment:
The question was specifically about live test.  Google credentials will fail on azure and vice versa.

Sent from my iThingy

> On Mar 29, 2016, at 16:05, Ignasi Barrera <notifications@github.com> wrote:
> 
> In apis/oauth/src/main/java/org/jclouds/oauth/v2/config/OAuthProperties.java:
> 
> > +    */
> > +   public static final String AZURE_CREDENTIAL = "oauth.azure-credential";
> > +
> > +   /**
> > +    * When using oauth with Azure Active Direction and Client Credentials, an "azure-identity"
> > +    * is the GUID associated with an application given permssion to perform operations in
> > +    * Azure Resource Manager.  Only used for OAuth Live Tests.
> > +    */
> > +   public static final String AZURE_IDENTITY = "oauth.azure-identity";
> > +
> > +   /**
> > +    * When using oauth with Azure Active Direction and Client Credentials, the azure-endpoint
> > +    * is the URL https://login.microsoftonline.com/<Tenant ID>/oauth2/token, where <Tenant ID>
> > +    * corresponds to your Azure AD Tenant.  Only used for OAuth Live Tests.
> > +    */
> > +   public static final String AZURE_ENDPOINT = "oauth.azure-endpoint";
> Following the discussion in the Azure PR, I don't think we need this Azure specific properties. We should be able to use the already existing identity and credential, and the oauth.endpoint property for this purpose.
> 
> Looking at the code, the value of the azure specific endpoint property is not injected in the authentication flow, so it is not used. If I'm not wrong, live tests just bind its value to the oauth.endpoint property to make it work, but that only tells the property is redundant. Same happens to the identity and credential properties. The authentication flow uses the Supplier<Credentials> (which provide the identity and credential configured when creating the context), so these properties don't take effect.
> 
> I'd say we could get rid of them and say that using Azure ARM just requires users to create the context using the App id and password as the credentials, and setting the oauth.endpoint property taking into account the tenant id?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>